packages/components/loading/src/directive.ts

```diff
@@ -1,56 +1,99 @@
-import Loading from './index'
+import { isRef, ref } from 'vue'
+import { Loading } from './service'
+import type { LoadingOptions } from './types'
+import type { LoadingInstance } from './loading'
+import type { Directive, DirectiveBinding, UnwrapRef } from 'vue'
 
-import type { DirectiveBinding } from 'vue'
-import type { ILoadingInstance } from './loading.type'
-
-const INSTANCE_NAME = 'ElLoading'
+const INSTANCE_KEY = Symbol('ElLoading')
 
+export type LoadingBinding = boolean | UnwrapRef<LoadingOptions>
 export interface ElementLoading extends HTMLElement {
-  [INSTANCE_NAME]?: ILoadingInstance
+  [INSTANCE_KEY]?: {
+    instance: LoadingInstance
+    options: LoadingOptions
+  }
 }
 
-const createInstance = (el: ElementLoading, binding: DirectiveBinding) => {
-  const textExr = el.getAttribute('element-loading-text')
-  const spinnerExr = el.getAttribute('element-loading-spinner')
-  const svgExr = el.getAttribute('element-loading-svg')
-  const svgViewBoxExr = el.getAttribute('element-loading-svg-view-box')
-  const backgroundExr = el.getAttribute('element-loading-background')
-  const customClassExr = el.getAttribute('element-loading-custom-class')
+const createInstance = (
+  el: ElementLoading,
+  binding: DirectiveBinding<LoadingBinding>
+) => {
   const vm = binding.instance
-  el[INSTANCE_NAME] = Loading({
-    text: (vm && vm[textExr]) || textExr,
-    svg: (vm && vm[svgExr]) || svgExr,
-    svgViewBox: (vm && vm[svgViewBoxExr]) || svgViewBoxExr,
-    spinner: (vm && vm[spinnerExr]) || spinnerExr,
-    background: (vm && vm[backgroundExr]) || backgroundExr,
-    customClass: (vm && vm[customClassExr]) || customClassExr,
-    fullscreen: !!binding.modifiers.fullscreen,
-    target: binding.modifiers.fullscreen ? null : el,
-    body: !!binding.modifiers.body,
-    visible: true,
-    lock: !!binding.modifiers.lock,
-  })
+
+  const getBindingProp = <K extends keyof LoadingOptions>(
+    key: K
+  ): LoadingOptions[K] =>
+    typeof binding.value === 'object' ? binding.value[key] : undefined
```

Comment:
**Suggestion:** can we use `isObject` under `@element-plus/utils/util` or `@vue/shared`?